easygcm/src/main/java/eu/inloop/easygcm/GcmHelper.java

```diff
@@ -185,30 +186,15 @@ void registerInBackground(final Context context, final RegistrationListener regi
         final AsyncTask<Void, Void, Void> registrationTask = new AsyncTask<Void, Void, Void>() {
```

Comment:
Look at the demo app here:
git clone https://github.com/googlesamples/google-services.git

They are using IntentService for performing the registrations. I think it's a good practice. Have a look at synchronization as well, the library should prevent multiple registrations at once.